prometheus_aws_guardduty_exporter/collector.py

```diff
@@ -39,7 +42,7 @@ def collect(self):
         return [currentFindingsMetric, scrapeErrorsMetric]
 
     def _collectMetricsByRegion(self, region):
-        client = boto3.client("guardduty", config=self.botoConfig, region_name=region)
+        client = self.botoClients[region]
```

Comment:
We were creating a new client for each request in order to have no issues with dirty state from previous client usage (ie. failures). What if we do the opposite: we continue instancing `boto3.client` in `_collectMetricsByRegion()` and we also instance the boto3 config in `_collectMetricsByRegion()` too?

Reply:
What kind of issues are you talking about? Can you give me an example?

Reply:
I remember in the past we saw something like a client fails (barely remember something related to keep alive connection stuck) and subsequent calls on the same client fails as well because of a dirty state. Unfortunately don't remember all details at this stage.